### ISSUE TITLE
[BUG] LAMMPS hangs for some cases with 'comm_style tiled' and 'balance rcb' commands with INTEL package and icc

### ISSUE THREAD

**Issue by mkanski** [lammps/lammps#3150]
**Summary**

Simulations hang at `run` when `comm_style tiled` and `balance rcb` commands are used for certain systems and number of threads.

EDIT:
I investigated this issue further and narrowed it down. **The problem only appears when INTEL package is installed and Intel compiler is used.**

The system with domains before and after balancing are below:

Before balance:
![before_balance](https://user-images.githubusercontent.com/20713012/155167892-7aab2af5-6089-4a08-9ea1-aee3f3f46649.png)
After balance:
![after_balance](https://user-images.githubusercontent.com/20713012/155167921-891a0d5b-1477-4930-844f-75fe74656dfd.png)

**LAMMPS Version and Platform**

Git info (develop / patch_7Jan2022-431-g6d2f434)
OS: Linux "CentOS Linux 7 (Core)" 3.10.0-862.14.4.el7.x86_64 x86_64

Compiler: Intel Classic C++ 20.21.1 / Intel(R) C++ g++ 6.5 mode with OpenMP 5.0 preview 1
C++ standard: C++11
MPI v3.1: Open MPI v4.1.0, package: Open MPI michal@HEAD Distribution, ident: 4.1.0, repo rev: v4.1.0, Dec 18, 2020

**Expected Behavior**

A standard run.

**Actual Behavior**

The simulations hang at `run 0` command. The problem does not occur when `comm_style brick` and `balance shift` commands are used instead, but the simulations are slower due to worse balance.

Partial stack trace is added in the last section. All but one threads are at `verlet.cpp:124` and the last one is left at `verlet.cpp:122`.

**Steps to Reproduce**

Run the following script with 20 threads. I can reproduce this problem on two other supercomputers, but I wasn't able to trigger it locally from my own computer (using `mpirun --oversubscribe`), so this issue may occur only for some combination of hardware and software.

EDIT:
I don't have Intel compilers installed on my computer, so I can't investigate this issue locally.

```
comm_style      tiled
region          u block -200 200 -200 200 -200 0
region          s sphere 0 0 0 100
create_box      1 u
lattice         sc 0.1
create_atoms    1 region s
mass            1 1
pair_style      lj/cut 6.0
pair_coeff      * * 1.0 1.0
balance         1.1 rcb
run             0
```

**Further Information, Files, and Links**

Stack trace (each two lines are from a different thread):

```
#8  0x0000000000796d97 in LAMMPS_NS::CommTiled::borders (this=0x3369e20) at lammps/src/comm_tiled.cpp:1333
#9  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x3369e20, flag=0) at lammps/src/verlet.cpp:124

#3  0x0000000000796d97 in LAMMPS_NS::CommTiled::borders (this=0x7feb5f6a2018) at lammps/src/comm_tiled.cpp:1333
#4  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7feb5f6a2018, flag=0) at lammps/src/verlet.cpp:124

#3  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x7fc98e423018) at lammps/src/comm_tiled.cpp:1242
#4  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7fc98e423018, flag=0) at lammps/src/verlet.cpp:124

#4  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x7f44f4124018) at lammps/src/comm_tiled.cpp:1242
#5  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7f44f4124018, flag=0) at lammps/src/verlet.cpp:124

#4  0x0000000000796d97 in LAMMPS_NS::CommTiled::borders (this=0x7f260000c018) at lammps/src/comm_tiled.cpp:1333
#5  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7f260000c018, flag=0) at lammps/src/verlet.cpp:124

#4  0x0000000000796d97 in LAMMPS_NS::CommTiled::borders (this=0x7f88405b5018) at lammps/src/comm_tiled.cpp:1333
#5  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7f88405b5018, flag=0) at lammps/src/verlet.cpp:124

#2  0x0000000000796d97 in LAMMPS_NS::CommTiled::borders (this=0x7f4469b91018) at lammps/src/comm_tiled.cpp:1333
#3  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7f4469b91018, flag=0) at lammps/src/verlet.cpp:124

#8  0x0000000000796d97 in LAMMPS_NS::CommTiled::borders (this=0x35f5b50) at lammps/src/comm_tiled.cpp:1333
#9  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x35f5b50, flag=0) at lammps/src/verlet.cpp:124

#8  0x0000000000796d97 in LAMMPS_NS::CommTiled::borders (this=0x25cab60) at lammps/src/comm_tiled.cpp:1333
#9  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x25cab60, flag=0) at lammps/src/verlet.cpp:124

#4  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x7f57c401f018) at lammps/src/comm_tiled.cpp:1242
#5  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7f57c401f018, flag=0) at lammps/src/verlet.cpp:124

#4  0x0000000000796d97 in LAMMPS_NS::CommTiled::borders (this=0x7f9ac001e018) at lammps/src/comm_tiled.cpp:1333
#5  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7f9ac001e018, flag=0) at lammps/src/verlet.cpp:124

#8  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x1aa0b50) at lammps/src/comm_tiled.cpp:1242
#9  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x1aa0b50, flag=0) at lammps/src/verlet.cpp:124

#3  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x7f148ec87018) at lammps/src/comm_tiled.cpp:1242
#4  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7f148ec87018, flag=0) at lammps/src/verlet.cpp:124

#3  0x0000000000796d97 in LAMMPS_NS::CommTiled::borders (this=0x7f2f74a43018) at lammps/src/comm_tiled.cpp:1333
#4  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7f2f74a43018, flag=0) at lammps/src/verlet.cpp:124

#4  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x7ff55b448018) at lammps/src/comm_tiled.cpp:1242
#5  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7ff55b448018, flag=0) at lammps/src/verlet.cpp:124

#7  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x22aab50) at lammps/src/comm_tiled.cpp:1242
#8  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x22aab50, flag=0) at lammps/src/verlet.cpp:124

#6  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x7fd5d36b2010) at lammps/src/comm_tiled.cpp:1242
#7  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x7fd5d36b2010, flag=0) at lammps/src/verlet.cpp:124

#8  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x2aa3e40) at lammps/src/comm_tiled.cpp:1242
#9  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x2aa3e40, flag=0) at lammps/src/verlet.cpp:124

#8  0x00000000007964b8 in LAMMPS_NS::CommTiled::borders (this=0x17a1e20) at lammps/src/comm_tiled.cpp:1242
#9  0x00000000005eed7d in LAMMPS_NS::Verlet::setup (this=0x17a1e20, flag=0) at lammps/src/verlet.cpp:124

#8  0x0000000000795438 in LAMMPS_NS::CommTiled::exchange (this=0x195ce30) at lammps/src/comm_tiled.cpp:996
#9  0x00000000005eed5a in LAMMPS_NS::Verlet::setup (this=0x195ce30, flag=0) at lammps/src/verlet.cpp:122


```

**Comment by sjplimp** [comment]
@mkanski Sorry for the long delay - looking at this bug report.  I don't understand the relevance of doing a build with the INTEL package.  Are you invoking the lj/cut/intel pair style with your launch command, e.g. via a -suffix command-line setting?
If not, I don't see how the any code in the INTEL package would be invoked.

You also say you run with 20 threads.  Do you mean you do an mpirun with -np 20 so that you run with 20 MPI tasks?
No threading involved via OpenMP and the INTEL package?

**Comment by mkanski** [comment]
@sjplimp Thanks for looking into this. However, I cannot replicate this issue anymore, with new version of LAMMPS and Intel compiler, so I'll close this bug report. 

To be honest, I don't remember any more details about this issue other than what is written here and that I used 20 MPI tasks.  I lost access to the computers which I used (and moved to a oneAPI Intel compiler), so I can't replicate the issue with old versions.
